test/tests/functional/pbs_maintenance_reservations.py

```diff
@@ -44,74 +44,62 @@ class TestMaintenanceReservations(TestFunctional):
     Two moms (-p "servers=M1,moms=M1:M2") are recommended for the tests.
     """
 
-    def setUp(self):
-        """
-        Tests setup
-        """
-        TestFunctional.setUp(self)
-
-        self.rsub_cmd = os.path.join(self.server.pbs_conf[
-            'PBS_EXEC'], 'bin', 'pbs_rsub')
-
-    @staticmethod
-    def now_plus_to_rsub_format(hours=0, minutes=0):
-        """
-        Function returns time now + hours + minutes in pbs_rsub
-        friendly format
-        """
-        interval = datetime.timedelta(hours=hours, minutes=minutes)
-        now_plus = datetime.datetime.today() + interval
-        return now_plus.strftime("%Y%m%d%H%M")
-
-    def submit_maintenance(self, user, start, end, hosts, others=""):
-        """
-        Submit maintenance reservation
-        """
-        cmd = "%s -R %s -E %s %s --hosts %s" % \
-              (self.rsub_cmd, start, end, others, " ".join(hosts))
-
-        return self.du.run_cmd(self.server.hostname, cmd, runas=user)
-
     def test_maintenance_acl_denied(self):
         """
         Test if the maintenance reservation is denied for common user
         """
-        start = self.now_plus_to_rsub_format(hours=1)
-        end = self.now_plus_to_rsub_format(hours=2)
+        now = int(time.time())
+
+        a = {'reserve_start': now + 3600,
+             'reserve_end': now + 7200}
+        h = [self.mom.shortname]
+        r = Reservation(TEST_USER, attrs=a, hosts=h)
 
-        ret = self.submit_maintenance(TEST_USER, start, end,
-                                      [self.mom.shortname])
+        msg = ""
 
-        msg = ret['err'][0].strip()
+        try:
+            self.server.submit(r)
+        except PbsSubmitError as err:
+            msg = err.msg[0].strip()
```

Comment:
Since maintenance reservations are handled in the pbs_rsub command, could you force PTL into CLI mode in setUp()?  This way if API mode is ever fixed, then this test won't fail.

Reply:
Done